waiter/test/waiter/service_description_test.clj

```diff
@@ -996,6 +1015,193 @@
+                :source-tokens (source-tokens-entry test-token token-data)}
+               result-map))))
+
+    (testing "exclusive service-mapping on token with on-the-fly header"
+      (let [token-data (assoc basic-token-data "service-mapping" "exclusive")
+            request-headers {"x-waiter-cpus" "2"}
+            result-map (run-computation token-data :request-headers request-headers)
+            service-description-template (-> token-data
+                                           (select-keys service-parameter-keys)
+                                           (assoc "cpus" 2 "permitted-user" current-user)
+                                           (assoc-in ["env" "WAITER_CONFIG_TOKEN"] test-token))]
+        (is (= {:component->previous-descriptor-fns {}
+                :core-service-description service-description-template
+                :on-the-fly? true
+                :reference-type->entry {:token {:sources (seq (source-tokens-entry test-token token-data))}}
+                :service-authentication-disabled false
+                :service-description (merge service-description-defaults service-description-template)
+                :service-id (service-description->service-id service-id-prefix service-description-template)
+                :service-preauthorized false
+                :source-tokens (source-tokens-entry test-token token-data)}
+               result-map))))
+
+    (testing "exclusive service-mapping on token with on-the-fly params header"
+      (let [token-data (assoc basic-token-data "allowed-params" #{"FEE" "FIE"} "service-mapping" "exclusive")
+            request-headers {"x-waiter-param-fee" "F1"
+                             "x-waiter-param-fie" "F2"}
+            result-map (run-computation token-data :request-headers request-headers)
+            service-description-template (-> token-data
+                                           (select-keys service-parameter-keys)
+                                           (assoc "env" {"FEE" "F1"
+                                                         "FIE" "F2"
+                                                         "WAITER_CONFIG_TOKEN" test-token}))]
+        (is (= {:component->previous-descriptor-fns {}
+                :core-service-description service-description-template
+                :on-the-fly? nil
+                :reference-type->entry {:token {:sources (seq (source-tokens-entry test-token token-data))}}
+                :service-authentication-disabled false
+                :service-description (merge service-description-defaults service-description-template)
+                :service-id (service-description->service-id service-id-prefix service-description-template)
+                :service-preauthorized true
+                :source-tokens (source-tokens-entry test-token token-data)}
+               result-map))))
+
+    (testing "exclusive service-mapping on token with run-as-requester"
```

Comment:
Haven't we determined that this shouldn't be allowed? It seems like we need to address this, and then have a test here specifically verifying that we don't accidentally re-enable `WAITER_CONFIG_TOKEN` when service-mapping=default.

Reply:
OK, discussed this with Shams. Rather than _disallowing_ this case, we're just going to provide the backend with additional information so that it can make informed decisions (e.g., regarding fetching tls certificates or other things that might be tricky with run-as-requester instances).